### ISSUE TITLE
Improve IrisInConfigurationSpaceTest.ConvexConfigurationSpace

### ISSUE THREAD

**Issue by jwnimmer-tri** [RobotLocomotion/drake#21031]
We noticed in #21029 that the test case `IrisInConfigurationSpaceTest.ConvexConfigurationSpace` effectively doesn't check anything right now.

(We also had to add a fudge factor for macOS builds to pass CI.)

**Comment by RussTedrake** [comment]
I don't believe that's accurate.  ConvexConfigurationSpace is a good test.  In #20974, I added a small additional check to that test in an attempt to cover the new `mixing_steps` option, and that new aspect of the test was insufficient. But the original test is still good.
